### PR TITLE
[AIR] Deprecate modules in `ray.tune.integration`

### PR DESCRIPTION
**Pull request by justinvyu**
## Why are these changes needed?

This PR deprecates the following:
1. Soft-deprecated, for removal in 2.7.
    - `ray.tune.integration.keras`, in favor of `ray.air.integrations.keras.ReportCheckpointCallback`.
2. Hard-deprecated, for removal in 2.6.
    - Passing wandb/mlflow API keys through config instead of the kwargs of `setup_wandb`/`setup_mlflow`. (Which was scheduled to be hard-deprecated in 2.4).
3. Removed (due to > 2 minor releases of hard-deprecation).
    -  `ray.tune.integration.comet` in favor of `ray.air.integrations.comet`

This PR also updates the API refs to reflect these changes. Namely, `ray-air/api/integrations` is now its own page instead of just linking to "Tune integrations".

See here:
- [New AIR integrations page](https://anyscale-ray--35069.com.readthedocs.build/en/35069/ray-air/api/integrations.html)
- [Condensed Tune integrations page](https://anyscale-ray--35069.com.readthedocs.build/en/35069/tune/api/integration.html)

### See also

- See the docs under "Deprecations" in [this doc](https://docs.google.com/document/d/1s2VouZg1CtzOCviZMHFFDPVMdJAPeXrTNcsBftwXmkQ/edit#) for context.
- Also see [this tracker spreadsheet](https://docs.google.com/spreadsheets/d/1rhMOGB1EwpEuLUu0BlBdzj28V2Wav96VqgnIIHJtLF8/edit).

If you don't have access to these docs, please feel free to leave a comment on this PR, and I can provide the context for these deprecations!

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tune/integration/keras.py [hunk 1]
```diff
@@ -65,6 +73,7 @@ def _handle(self, logs: Dict, when: str = None):
         tune.report(**report_dict)
 
 
+@Deprecated(message=_DEPRECATION_MESSAGE)
 class _TuneCheckpointCallback(TuneCallback):
     """Keras checkpoint callback
```
Comment (matthewdeng) [inline python/ray/tune/integration/keras.py:78]: Any reason for annotating this nonpublic class?

File: python/ray/air/callbacks/keras.py [hunk 2]
```diff
@@ -1,4 +0,0 @@
-from ray.tune._structure_refactor import warn_structure_refactor
-from ray.air.integrations.keras import *  # noqa: F401, F403
-
-warn_structure_refactor(__name__, "ray.air.integrations.keras")
```
Comment (matthewdeng) [inline python/ray/air/callbacks/keras.py:4]: @krfricke did we ever establish a cleanup policy for `warn_structure_refactor`? Could we clean up the rest of these as well (in a separate PR)?

File: python/ray/tune/examples/wandb_example.py
Comment (matthewdeng) [inline python/ray/tune/examples/wandb_example.py:1]: Is there a replacement for this?

Comment (justinvyu) [reply python/ray/tune/examples/wandb_example.py:1]: It's `tune-wandb.ipynb`, which is already the only thing shown in docs.

File: doc/source/ray-air/api/integrations.rst [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+
+Ray AIR Integrations
+====================
+
+.. currentmodule:: ray
+
+.. _air-monitoring-integrations:
+.. _air-builtin-callbacks:
+
+Experiment Monitoring Integrations
```
[the remark this answers is not in the document]
Comment (justinvyu) [reply doc/source/ray-air/api/integrations.rst:11]: Yeah user guide makes sense.

There are a few things that need a spot in the API references though.

What about: keeping 1 page of monitoring and ml integrations, with minimal text - just API ref links.

Then, add an AIR user guide with the X ways of training with framework Y.

File: python/ray/tune/integration/keras.py (hunk 1 above)
Comment (justinvyu) [reply python/ray/tune/integration/keras.py:78]: No reason really, felt like it's nice to tag for deletion later.

File: doc/source/ray-air/api/integrations.rst [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+**Usage Scenario:** Non-distributed training, where the dataset is relatively small
+and there are many trials (e.g., many hyperparameter configurations).
+Use vanilla PyTorch with Ray Tune to parallelize model training.
+
+.. seealso::
+
+    :ref:`See an example here. <tune-pytorch-cifar-ref>`
+
+
+|:two:| ``TorchTrainer``
```
Comment (krfricke) [inline doc/source/ray-air/api/integrations.rst:86]: Should we link this (make it a ref?)

File: doc/source/ray-air/api/integrations.rst [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+.. autosummary::
+
+    ~train.xgboost.XGBoostTrainer
+
+.. seealso::
+
+    :ref:`See an example here. <air-xgboost-example-ref>`
+
+
+|:three:| ``xgboost_ray``
```
Comment (krfricke) [inline doc/source/ray-air/api/integrations.rst:225]: Link to repo?

**Review by krfricke — APPROVED** [review]
Deprecations LGTM, pinging @maxpumperla for docs structure review. There seems to be a conflict that needs to be resolved before merge

File: doc/source/ray-air/api/experiment-tracking.rst [hunk 6]
```diff
@@ -0,0 +1,51 @@
+.. currentmodule:: ray
+
+.. _air-experiment-tracking-integrations:
```
Comment (maxpumperla) [inline doc/source/ray-air/api/experiment-tracking.rst:3]: General comment: https://docs.ray.io/en/latest/tune/examples/index.html#tune-experiment-tracking-examples we have a couple of examples here, not sure if we covered all of them in this migration yet.

File: doc/source/ray-air/api/integrations.rst [hunk 7]
```diff
@@ -0,0 +1,230 @@
...
+There are 2 recommended ways to train PyTorch models on a Ray cluster.
+
+.. note::
+
+    If you're training PyTorch models with PyTorch Lightning, see :ref:`below <air-pytorch-lightning>`
+    for the available PyTorch Lightning Ray AIR integrations.
+
+See the options |:one:| |:two:| below, along with the usage scenarios and API references for each:
+
+|:one:| Vanilla PyTorch with Ray Tune
```
Comment (maxpumperla) [inline doc/source/ray-air/api/integrations.rst:20]: I think this could be a `tabs` section to compress things a bit (later sections, too)

**Review by maxpumperla — APPROVED** [review]
Agree with the changes. We can adjust doc placement etc. later on. But the way you worded things, we should see good results for this e.g. in Algolia search.

File: doc/source/ray-air/api/integrations.rst (hunk 7 above)
Comment (justinvyu) [reply doc/source/ray-air/api/integrations.rst:20]: I tried this out and it doesn't let me add links as the tab headers -- do you think it's ok as is?

**Pull request merged**